test/integration/render-tests/collator/default/style.json

```diff
@@ -0,0 +1,225 @@
+              "type": "Point",
+              "coordinates": [
+                92.8125,
+                72.39570570653261
+              ]
+            }
+          },
+          {
+            "type": "Feature",
+            "properties": {
+                "a": "a",
+                "b": "ä",
+                "operator": "<",
+                "case-sensitive": false,
+                "diacritic-sensitive": true
+            },
+            "geometry": {
+              "type": "Point",
+              "coordinates": [
+                94.21875,
+                34.30714385628804
+              ]
+            }
+          },
+          {
+            "type": "Feature",
+            "properties": {
+                "a": "a",
+                "b": "A",
+                "operator": "<=",
+                "case-sensitive": true,
+                "diacritic-sensitive": true
+            },
+            "geometry": {
+              "type": "Point",
+              "coordinates": [
+                97.03125,
+                -44.087585028245165
+              ]
+            }
+          }
+        ]
+      }
+    }
+  },
+  "glyphs": "local://glyphs/{fontstack}/{range}.pbf",
+  "layers": [
+    {
+      "id": "background",
+      "type": "background",
+      "paint": {
+        "background-color": "white"
+      }
+    },
+    {
+      "id": "literal",
+      "type": "symbol",
+      "source": "mapbox",
+      "layout": {
+        "text-field": ["let",
+            "a", ["to-string", ["get", "a"]],
+            "b", ["to-string", ["get", "b"]],
+            "op", ["get", "operator"],
+            "collation", ["collator", {
+                "case-sensitive": ["get", "case-sensitive"],
+                "diacritic-sensitive": ["get", "diacritic-sensitive"],
+                "locale": "en" }
+            ],
+            ["concat",
+                ["var", "a"], " ",
+                ["var", "op"], " ",
+                ["var", "b"], "\n",
+                "Case: ", ["to-string", ["get", "case-sensitive"]], "\n",
+                "Diacritics: ", ["to-string", ["get", "diacritic-sensitive"]], "\n",
+                ["to-string", ["case",
+                    ["==", ["var", "op"], "=="],
+                    ["==", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    ["==", ["var", "op"], "!="],
+                    ["!=", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    ["==", ["var", "op"], "<"],
+                    ["<", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    ["==", ["var", "op"], "<="],
+                    ["<=", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    ["==", ["var", "op"], ">"],
+                    [">", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    ["==", ["var", "op"], ">="],
+                    [">=", ["var", "a"], ["var", "b"], ["var", "collation"]],
+                    false
+                ]]
```

Comment:
This is a nice test expression! :)

Nit: could it be a little simpler with `match`?

```js
                ["to-string", ["match",
                    ["var", "op"],
                    "==", ["==", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    "!=", ["!=", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    "<", ["<", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    "<=", ["<=", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    ">", [">", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    ">=", [">=", ["var", "a"], ["var", "b"], ["var", "collation"]],
                    false
                ]]
```

Reply:
Good idea! Yeah it was actually nice to be forced to think a little bit about a non-trivial collator expression.